File: tournaments/arrayMaxConsecutiveSum/arrayMaxConsecutiveSum.cs
int arrayMaxConsecutiveSum(int[] inputArray, int k) {
    var result = int.MinValue;
    var sum = 0;
    
    for (int i = 0; inputArray.Length > i; ++i) {
        if (k <= i) {
            sum -= inputArray[i - k];
        }
        
        sum += inputArray[i];
        result = Math.Max(result, sum);
    }
    
    return result;
}

File: tournaments/arrayMaxConsecutiveSum/arrayMaxConsecutiveSum.cpp
int arrayMaxConsecutiveSum(std::vector<int> inputArray, int k) {

  int result = 0,
      currentSum = 0;

  for (int i = 0; i < k - 1; i++) {
    currentSum += inputArray[i];
  }
  for (int i = k - 1; i < inputArray.size(); i++) {
    currentSum += inputArray[i];
    if (currentSum > result) {
      result = currentSum;
    }
    currentSum -= inputArray[i - k + 1];
  }

  return result;
}
